Game/MyProject/Game/SceneManagement/SceneNode.cpp

```diff
@@ -159,9 +173,23 @@ void SceneNode::setAnimator(const std::shared_ptr<NodeAnimator>& animator) {
     _animator = animator;
 }
 
+// returns the animator
+const NodeAnimator* SceneNode::getAnimator() const {
+    return _animator.get();
+}
+
 // sets the mesh instance
 void SceneNode::setRenderable(const std::shared_ptr<MeshInstance>& renderable) {
     _rendarable = renderable;
 }
 
+void SceneNode::setIsMoving(bool value) {
+    _is_moving = value;
+}
+
+// returns true is the node is moving
+bool SceneNode::getIsMoving() {
+    return _is_moving;
```

Comment:
ideal case is to ask the animator if the node is moving. however you can keep this as an optimization.